lib/components/search/Search.js

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import AsyncSelect from 'react-select/lib/Async';
+import { withStyles } from '@material-ui/core/styles';
+
+import Placeholder from './Placeholder';
+import NoOptionsMessage from './NoOptionsMessage';
+import Control from './Control';
+import SingleValue from './SingleValue';
+import ValueContainer from './ValueContainer';
+import DropdownIndicator from './DropdownIndicator';
+import Menu from './Menu';
+import OptionContainer from './OptionContainer';
+import GroupHeading from './GroupHeading';
+import Group from './Group';
+
+const styles = theme => ({
+  root: {
+    position: 'relative',
+  },
+});
+
+const IndicatorSeparator = () => null;
+
+class Search extends React.Component {
+  render() {
+    const {
+      classes,
+      theme,
+      optionComponent: OptionComponent,
+      onInputChange,
+      onSelectOption,
+      value,
+      placeholder,
+    } = this.props;
+
+    const selectStyles = {
+      input: base => ({
+        ...base,
+        color: theme.palette.text.primary,
+      }),
+    };
+
+    const Option = props => (
```

Comment:
This component could be declared outside the component. What do you think?

Reply:
Cannot do this because `Option` depends on the `optionComponent` prop.

Reply:
Ah, overlooked that.